### PR TITLE
Ensure pushStatus is properly running

### PR DESCRIPTION
**Pull request by dplewis**
### New Pull Request Checklist

- [ ] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [ ] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

This fixes a flaky test. 

```
PushController properly creates _PushStatus
Message:
Expected 'pending' to be 'running'.
Stack:
Error: Expected 'pending' to be 'running'.
at
at reconfigureServer.then.then.then.then.then.catch.then (/home/runner/work/parse-server/parse-server/spec/PushController.spec.js:618:48)
at process._tickCallback (internal/process/next_tick.js:68:7)
```

This issue is the status of a push doesn't get updated properly. Here are the push status.

1) pending - initial state
2) running - set Running
3) running - still running, updated counts, numSent, numFailed etc.
4) succeeded - final state

The test fails because the status in step 2 never gets updated to `running` and as a result step 3 is never set to `running`.

### Approach

Ensure status is set to running if you are keeping track of pushes sent, failed etc. I don't know if there is a bigger issue here because it looks like the same is happening to the Job status flaky tests. This is fixing a potential server bug.

### TODOs before merging

- [ ] Add test cases
- [ ] Add entry to changelog
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK 
- [ ] ...

### CODE REVIEW

**Comment by mtrezza** [comment]
> The test fails because the status in step 3 goes from running back to pending.

How does it go from `running` back to `pending`? Is there a specific line that causes that?

**Comment by dplewis** [comment]
Sorry looking at it again. As @davimacedo said in https://github.com/parse-community/parse-server/pull/7196 there is an error that occurs when updating a status that was just created. So maybe both 2 and 3 are still pending and never got set to running.

**Comment by mtrezza** [comment]
The conclusion in https://github.com/parse-community/parse-server/pull/7196#issuecomment-781538789 was that the test design caused the flakiness. In this PR it seems that the server code has been changed, that's why I'm wondering.

File: spec/PushController.spec.js [hunk 1]
```diff
@@ -79,19 +79,6 @@ describe('PushController', () => {
     done();
   });
 
-  it('can throw on validateDeviceType when single invalid device type is set', done => {
```
Comment (dplewis) [inline spec/PushController.spec.js:82]: I'm deleting this test as it already exists

Screen Shot 2021-02-20 at 2 52 24 PM

**Comment by dplewis** [comment]
> In this PR it seems that the server code has been changed, that's why I'm wondering.

Thats whats interesting nobody would notice in this in production. If the state went from pending -> succeeded. This PR fixes a potential server bug and a flaky test.

**Review by mtrezza — APPROVED** [review]
Coolio!

**Comment by mtrezza** [comment]
I'm adding this fail to the list "Parse.GeoPoint testing creating geo point exception two fields".

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**